e2etest/zt_newe2e_fns_dir_test.go

```diff
@@ -202,3 +202,168 @@ func (*FNSSuite) Scenario_SyncTrailingSlashDeletion(a *ScenarioVariationManager)
 		},
 	}, false)
 }
+
+func (*FNSSuite) Scenario_SyncOverlap(a *ScenarioVariationManager) {
+	// Sync must be capable of mirroring a source resource type onto a blob destination.
+
+	// Define our scenario up front; reduce complexity in debugging.
+	srcLoc := ResolveVariation(a, []common.Location{common.ELocation.File(), common.ELocation.Blob(), common.ELocation.BlobFS(), common.ELocation.Local()})
+	dstLoc := common.ELocation.Blob()
+	a.InsertVariationSeparator("->Blob|Overwrite:")
+
+	const (
+		DestTypeStandardStub = iota
+		DestTypeOverlapStup
+	)
+
+	dstType := NamedResolveVariation(a, map[string]uint{
+		"StandardStub":      DestTypeStandardStub,
+		"TrailingSlashStub": DestTypeOverlapStup,
+	})
+
+	// Set up the resource maps
+	srcMap := ObjectResourceMappingFlat{
+		"foo": ResourceDefinitionObject{},
+	}
+
+	dstMap := map[uint]ObjectResourceMappingFlat{
+		DestTypeStandardStub: {
+			"foo": ResourceDefinitionObject{
+				ObjectProperties: ObjectProperties{
+					EntityType: common.EEntityType.Folder(),
+				},
+			},
+		},
+		DestTypeOverlapStup: {
+			"foo/": ResourceDefinitionObject{
+				ObjectProperties: ObjectProperties{
+					EntityType: common.EEntityType.Folder(),
+				},
+			},
+		},
+	}[dstType]
+
+	// Spin up the resources
+	src := CreateResource(a, GetRootResource(a, srcLoc), ResourceDefinitionContainer{
+		Objects: srcMap,
+	})
+
+	dst := CreateResource(a, GetRootResource(a, dstLoc), ResourceDefinitionContainer{
+		Objects: dstMap,
+	})
+
+	// Execute the test
+	RunAzCopy(a, AzCopyCommand{
+		Verb: AzCopyVerbSync,
+		Targets: []ResourceManager{
+			src.GetObject(a, "foo", common.EEntityType.File()),
+			dst.GetObject(a, "foo", common.EEntityType.File()),
+		},
+		Flags: SyncFlags{
+			CopySyncCommonFlags: CopySyncCommonFlags{
+				Recursive:             pointerTo(true),
+				IncludeDirectoryStubs: pointerTo(true),
+			},
+		},
+	})
+
+	ValidateResource(a, dst, ResourceDefinitionContainer{
+		Objects: ObjectResourceMappingFlat(JoinMap(dstMap, srcMap)),
+	}, false)
+}
+
+/*
+This tests that sync is capable of creating the destination resource if it does not exist, including the root container.
+*/
+//func (*FNSSuite) Scenario_OverlapWithTrailingSlash(a *ScenarioVariationManager) {
```

Comment:
Should this be uncommented?

Reply:
It should be deleted, it was leftover from a prior iteration of testing on this PR.